harness/src/test/java/org/jboss/pvt/harness/TestConfiguration.java

```diff
@@ -63,10 +110,80 @@ public Properties getAllConfiguration()
      * @return a new statement, which may be the same as {@code base},
      *         a wrapper around {@code base}, or a completely new Statement.
      */
-    @Override
-    public Statement apply( Statement base, Description description )
-    {
-//        throw new PVTSystemException( "NYI" );
-        return base;
+    public Statement apply(Statement base, Description description) {
+        return statement(base);
+    }
+
+    private Statement statement(final Statement base) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                before();
+                try {
+                    base.evaluate();
+                } finally {
+                    after();
+                }
+            }
+        };
+    }
+
+    /**
+     * Override to set up your specific external resource.
+     *
+     * @throws Throwable if setup fails (which will disable {@code after}
+     */
+    protected void before() throws Throwable {
+        // load properties
+        if(properties.isEmpty()) {
+
+            try {
+                properties.load(new FileReader(PROPERTIES_CONFIG_FILE));
+            } catch (Exception e) {
+                throw new PVTSystemException("File to load config file: " + PROPERTIES_CONFIG_FILE, e);
+            }
+        }
+        prepareResources();
+    }
+
+    /**
+     * Override to tear down your specific external resource.
+     */
+    protected void after() {
+        // do nothing
+    }
+
+    protected synchronized void prepareResources() throws Exception{
+
+        if(distribution == null) {
+            String distributionUrl = getConfiguration(PRODUCT_ZIP_URL_KEY);
```

Comment:
What happens if you want to validate various types of zip files. You are limited to a single zip, that has to have a large download and can't be altered.